### PR TITLE
chore: fallback and adding retry

### PR DESCRIPTION
**Pull request by davidmao2**
Solved the MaxRetry issue.

### CODE REVIEW

File: src/datadog.py [hunk 1]
```diff
@@ -114,6 +114,9 @@ def load_metrics(self, queried: bool, filter_prefix: str = None) -> [Metric]:
     def update_metric_tags(self, metric: Metric, tags: [str], aggregations: [MetricCustomAggregation]):
         aggregations = _fix_metrics_custom_aggregations(metric, aggregations)
 
+        if metric.metadata.type is None: #Temporary fallback
+            metric.metadata.type = "distribution"
+
```
[the remark this answers is not in the document]
Comment (Fs02) [reply src/datadog.py:119]: is this works? since you still have to remove tag manually right?

File: src/slack.py [hunk 2]
```diff
@@ -1,10 +1,22 @@
...
+from urllib3.util.retry import Retry
+from requests.adapters import HTTPAdapter
 
 def notify(text=str, attachments: [dict] = None):
     url = os.getenv("SLACK_WEBHOOK_URL")
     if url is None:
         return
 
-    requests.post(url, json={"text": text, "attachments": attachments})
+    session = requests.Session()
```
Comment (Fs02) [inline src/slack.py:11]: seems this and adapter can be put outside function? so session not created on every requests

File: src/datadog.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (davidmao2) [reply src/datadog.py:119]: I think should work, after I changed the tag manaually, but still the type is `null`( has no changed).

File: src/slack.py (hunk 2 above)
Comment (davidmao2) [reply src/slack.py:11]: Updated

Comment (Fs02) [reply src/slack.py:11]: seems no changes?

Comment (davidmao2) [reply src/slack.py]: Updated once..somehow it didn't push to remote.

File: src/datadog.py [hunk 3]
```diff
@@ -138,31 +145,38 @@ def touch_metric_metadata(self, metric: Metric):
...
 
     def _update_tag_configuration(self, metric: Metric, tags: [str], aggregations: [MetricCustomAggregation]):
-        update_tag_request = MetricTagConfigurationUpdateRequest(
-            data=MetricTagConfigurationUpdateData(
-                id=metric.metric_id,
-                type=MetricTagConfigurationType.MANAGE_TAGS,
-                attributes=MetricTagConfigurationUpdateAttributes(
-                    tags=tags,
-                    aggregations=aggregations,
+        try:
```
Comment (Fs02) [inline src/datadog.py:165]: committed by mistake?

File: src/main.py [hunk 4]
```diff
@@ -52,13 +52,14 @@ def optimize_inactive_metrics(metric: Metric):
 
 
 def main():
-    filter_prefix = "sn_starter"
+    filter_prefix = "sn_feed"
```
Comment (Fs02) [inline src/main.py:55]: also committed by mistake

Comment (davidmao2) [reply src/main.py]: Thanks for catching, just removed the main.py changes.

File: src/datadog.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (davidmao2) [reply src/datadog.py:119]: I see, shall we keep it? Or I can remove.

File: src/datadog.py (hunk 3 above)
Comment (davidmao2) [reply src/datadog.py]: Removed the changes

File: src/datadog.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Fs02) [reply src/datadog.py:119]: yes, you can remove it

let me handle this in other PR

**Review by Fs02 — APPROVED** [review]
LGTM

**Pull request merged**